spec/components/collections/edit_license_component_spec.rb

```diff
@@ -33,6 +33,71 @@
     end
   end
 
+  context "with custom rights for the collection disallowed" do
+    let(:collection) { build(:collection, allow_custom_rights_statement: false) }
+
+    it "selects the 'No' option for 'Include custom use statement'" do
+      expect(rendered.to_html).to match(/<input .*checked name="allow_custom_rights_statement" id="allow_custom_rights_statement_false"/)
+    end
+  end
+
+  # TODO: likely won't solve your test issue, which seems reform/factorybot/model related, but can do e.g. expect(rendered.css("a").map { |node| node["aria-label"] }).to include "Copy persistent link to #{WorkTitlePresenter.show(work.head)}"
+  # instead of matching text in HTML, which is always a PITA and fragile.  see e.g. https://github.com/sul-dlss/happy-heron/pull/3219/files#diff-25aaa01f872eae7566879d2538b865eb517a7a486ac65f8cd5af6deebcb26e6bR126-R128
+  # TODO: i wonder if something like this would help for my mis-behaving collection fixture: https://github.com/sul-dlss/happy-heron/pull/3235/files#diff-9a72d569948126ae4b4d04da9e29997e9cb9320767717942e47e02cee56e52aeR14
+  # e.g. collection_version.collection.update!(head: collection_version)
+  context "with custom rights for the collection allowed" do
+    let(:collection) { build(:collection, allow_custom_rights_statement: true) }
+
+    it "selects the 'Yes' option for 'Include custom use statement'" do
+      expect(rendered.to_html).to match(/<input .*checked name="allow_custom_rights_statement" id="allow_custom_rights_statement_true"/)
+    end
+
+    context "with custom rights provided by the collection" do
+      # let(:collection) { build(:collection, :with_custom_rights_from_collection) }
+      let(:collection) { create(:collection, :with_custom_rights_from_collection) }
+      let(:collection_version) { create(:collection_version, collection:) }
+
+      before { collection_version.collection.head = collection_version }
+
+      # TODO: this hack didn't work.  should get this through model methods acting on factorybot defined fields anyway.
+      # before { allow(collection).to receive(:custom_rights_statement_source_option).and_return("provided_by_collection") }
+
+      it "selects the 'Provide specific terms' option" do
+        # byebug
```

Comment:
Here's what I find most perplexing at the moment about this test: if I uncomment this `byebug` call, and drop into the debugger, I see something like:
```
(byebug) collection.head == collection_version
true
(byebug) collection_form.collection == collection
true
(byebug) collection.custom_rights_statement_source_option
"provided_by_collection"
(byebug) collection_form.collection.custom_rights_statement_source_option
"provided_by_collection"
(byebug) collection&.custom_rights_statement_source_option
"provided_by_collection"
(byebug) 
```

The thing i find perplexing there is that `collection.custom_rights_statement_source_option` is `"provided_by_collection"` (what I'd expect given the fixture setup and the logic of `Collection#custom_rights_statement_source_option`).  And all the other associations seem correct.  But then it would seem that `EditLicenseComponent` isn't getting that value from the `Collection` object in the test or something, because the radio button that's examined by this test isn't being selected according to the value in the collection.  But it does work in the actual form when I test manually by creating or editing a collection in my local H2 instance.

When I was asking for debugging advice on this before last week's FR shift, @justinlittman suspected that the thing that was breaking the existing unit tests was that the `custom_rights_statement_source_option` property in `CollectionSettingsForm` was `writeable: true`.  Changing it to be `virtual: true` as it is now in this branch (i.e. not writeable or readable from the model, [see here](https://trailblazer.to/2.1/docs/reform.html#virtual)) unbroke existing tests, but it does seem fishy to me that the misbehaving property here is `virtual`, and that the tests for values in `textarea` elements (which are not virtual and come directly from the model) are passing.  One of Justin's suggestions was a re-thinking of the data model so that there's an actual `custom_rights_statement_option` field with values like `no_custom`, `provided_by_collection`, and `entered_by_depositor` (instead of the current `allow_custom_rights_statement` boolean and some logic acting on that and the presence of e.g. custom instructions text).  I initially went for the half-measure seen in this branch, where I kept the boolean but got rid of `writeable`, because I figured it'd be a more expedient way to get what I had working, and then I could revisit the data model change once I had the tests working with something closer to the approach I'd already started on.  But now I think I'll just try taking that suggestion in full.

If I follow that path, I think it'll mean that the "Yes" radio button is still a virtual property in the Reform code, and so maybe it'll be hard to test, but all of the other radio buttons should then become more testable, if that approach pays off.